Sources/NIO/TypeAssistedChannelHandler.swift

```diff
@@ -77,4 +77,4 @@ extension ChannelOutboundHandler {
 }
 
 /// A combination of `ChannelInboundHandler` and `ChannelOutboundHandler`.
-public protocol ChannelDuplexHandler: ChannelInboundHandler, ChannelOutboundHandler { }
+public typealias ChannelDuplexHandler = ChannelInboundHandler & ChannelOutboundHandler
```

Comment:
@ianpartridge thats actually a very nice idea I did not even think of... @Lukasa @weissi WDYT ?

Reply:
No strong preference, tbh, I’ll defer to @weissi.

Reply:
FWIW: One downside is that you won't be able to write `extension ChannelDuplexHandler`.

Reply:
Also error messages will be different I guess. I’d be inclined to merge this PR as we can always (without breaking api) go back to a separate protocol.